src/core/reactor_backend.cc

```diff
@@ -331,32 +330,41 @@ aio_general_context::~aio_general_context() {
 }
 
 void aio_general_context::queue(linux_abi::iocb* iocb) {
-    assert(last < end);
-    *last++ = iocb;
+    if (iocbs.size() < iocbs.capacity()) {
+        iocbs.push_back(iocb);
+    } else {
+        iocbs_backlog.push_back(iocb);
+    }
 }
 
 size_t aio_general_context::flush() {
-    auto begin = iocbs.get();
-    using clock = std::chrono::steady_clock;
-    constexpr clock::time_point no_time_point = clock::time_point(clock::duration(0));
-    auto retry_until = no_time_point;
+    auto begin = iocbs.data();
+    auto last = iocbs.data() + iocbs.size();
     while (begin != last) {
         auto r = io_submit(io_context, last - begin, begin);
-        if (__builtin_expect(r > 0, true)) {
```

Comment:
I guess one question is if there is any scenario where retrying worked before? I.e., is it possible to get EAGAIN at some point in the loop then after trying for a while, successfully submit some iocbs?

Reply:
Yes, you are right, I'm not even sure where I was originally going with that line of thought.